deebotozmo/cli.py

```diff
@@ -0,0 +1,458 @@
+            print(f"Stats Type: {event.type}")
+            lock.set()
+
+        lock.clear()
+        listener = vacbot.bot.events.stats.subscribe(on_stats_event)
+        await lock.wait()
+        listener.unsubscribe()
+    finally:
+        await vacbot.after()
+
+
+@cli.command(help="Get robot components life span")
+@coro
+async def components() -> None:
+    """Click subcommand that returns the robot's life span."""
+    vacbot = CliUtil()
+    try:
+        await vacbot.before()
+
+        lock = asyncio.Event()
+
+        async def on_lifespan_event(event: dict) -> None:
+            for key, value in event.items():
+                print(f"{key}: {value}%")
+            lock.set()
+
+        lock.clear()
+        listener = vacbot.bot.events.lifespan.subscribe(on_lifespan_event)
+        await lock.wait()
+        listener.unsubscribe()
+    finally:
+        await vacbot.after()
+
+
+@cli.command(name="getrooms", help="Get saved rooms")
+@coro
+async def get_rooms() -> None:
+    """Click subcommand that returns saved room."""
+    vacbot = CliUtil()
+    try:
+        await vacbot.before()
+
+        lock = asyncio.Event()
+
+        async def on_rooms(event: RoomsEvent) -> None:
+            for room in event.rooms:
+                print(f"{room.id} {room.subtype}")
+            lock.set()
+
+        lock.set()
+        listener = vacbot.bot.map.events.rooms.subscribe(on_rooms)
+        await lock.wait()
+        listener.unsubscribe()
+    finally:
+        await vacbot.after()
+
+
+@cli.command(
+    name="exportlivemap",
+    help='Get robot map and save it [filepath ex: "/folder/livemap.png"',
+)
+@click.argument("filepath", type=click.STRING, required=True)
+@click.option(
+    "--force-extension/--no-force-extension",
+    default=False,
+    help="Do not check if the file extension is valid.",
+)
+@coro
+async def export_live_map(filepath: str, force_extension: bool) -> None:
+    """Click subcommand that returns the live map."""
+    if not force_extension and mimetypes.guess_type(filepath)[0] != "image/png":
+        logging.error("exportlivemap generates a png image.")
+        logging.error(
+            "either change your file extension to 'png' or pass '--force_extension'."
+        )
+        return
+
+    vacbot = CliUtil()
+    try:
+        await vacbot.before()
+
+        lock = asyncio.Event()
+
+        async def on_map(_: MapEvent) -> None:
+            with open(filepath, "wb") as file:
+                file.write(base64.decodebytes(vacbot.bot.map.get_base64_map()))
+            lock.set()
+
+        lock.clear()
+        listener = vacbot.bot.events.map.subscribe(on_map)
+        await lock.wait()
+        listener.unsubscribe()
+    finally:
+        await vacbot.after()
+
+
+class CliUtil:
+    """CliUtil communicates with the bot to provide self.bot."""
+
+    def __init__(self) -> None:
+        """Init CliUtil."""
+        self._config = read_config()
+
```

Comment:
```suggestion
    self.bot: Optional[VacuumBot] = None
```

Reply:
Would do you mean?

Reply:
I think you mean it as a solution to fix pylint error: https://github.com/And3rsL/Deebotozmo/pull/62/commits/885fe25f4ec4642a05d45eb3147e085199f7fc29

Reopen if I'm wrong :)

Reply:
Still there

Reply:
Doing this causes `mypy` to complain:
```
deebotozmo/cli.py:192: error: Item "None" of "Optional[VacuumBot]" has no attribute "execute_command"  [union-attr]
deebotozmo/cli.py:307: error: Item "None" of "Optional[VacuumBot]" has no attribute "events"  [union-attr]
deebotozmo/cli.py:331: error: Item "None" of "Optional[VacuumBot]" has no attribute "events"  [union-attr]
deebotozmo/cli.py:340: error: Item "None" of "Optional[VacuumBot]" has no attribute "events"  [union-attr]
deebotozmo/cli.py:349: error: Item "None" of "Optional[VacuumBot]" has no attribute "events"  [union-attr]
deebotozmo/cli.py:358: error: Item "None" of "Optional[VacuumBot]" has no attribute "events"  [union-attr]
deebotozmo/cli.py:385: error: Item "None" of "Optional[VacuumBot]" has no attribute "events"  [union-attr]
deebotozmo/cli.py:409: error: Item "None" of "Optional[VacuumBot]" has no attribute "events"  [union-attr]
deebotozmo/cli.py:433: error: Item "None" of "Optional[VacuumBot]" has no attribute "map"  [union-attr]
deebotozmo/cli.py:471: error: Item "None" of "Optional[VacuumBot]" has no attribute "map"  [union-attr]
```

It does fix pylint error though

Reply:
I fixed it by accident without using `assert isinstance...` or anything. Not sure if it is a bug in the checker but https://github.com/And3rsL/Deebotozmo/pull/62/commits/8e6f3d0f99cea7a3e54da8a9f61d03f2ec863130 fixed it